### PR TITLE
refactor entitlement proposal to apply to pods and builds

### PR DESCRIPTION
**Pull request by bparees**
also remove obsolete info

### CODE REVIEW

**Comment by bparees** [comment]
@gabemontero here's the refactor PR i promised.

File: enhancements/builds/entitlements.md
Comment (gabemontero) [inline enhancements/builds/entitlements.md]:
```suggestion
MCM also introduces a "compliance" with a list of "policies" (all backed by CRDs) which the klusterlet will
```

File: enhancements/builds/entitlements.md
Comment (gabemontero) [inline enhancements/builds/entitlements.md]:
```suggestion
do not have or ever will the entire set of subscription manager RPMs, and as discussed in some of the alternatives
```

File: enhancements/builds/entitlements.md
Comment (gabemontero) [inline enhancements/builds/entitlements.md]:
```suggestion
1. We did *NOT* provide for multiple sets of entitlement credentials in 3.x.  There have been no official, RFE style registered requirements entitlements at a namespace level.  However, we've started to get (slightly) less official feedback that customers want this in a more automated sense (vs. the current manual steps).  At a minimum, and is noted below when alternatives are discussed, if namespaced entitlements are added with a subsequent implementation after global entitlements usability is improved, it needs to be seamless, and we minimally need to agree now on what the order of precedence will be (with the take on that listed below).
```

Comment (gabemontero) [reply enhancements/builds/entitlements.md]: in case it is not clear, I change `with subsequent implementation` to `with a subsequent implementation`

File: enhancements/builds/entitlements.md
Comment (gabemontero) [inline enhancements/builds/entitlements.md]:
```suggestion
This enhancement proposes a new concept of a "global secret".  Administrators will be able to define secrets in a unique namespace (likely openshift-config) and then configure a CSI driver + Admission controller which will be able to inject those secrets into pods that request the content, based on a configuration the administrator provides.
```

File: enhancements/builds/entitlements.md
Comment (gabemontero) [inline enhancements/builds/entitlements.md:212]: Similar to your suggestion in https://github.com/openshift/cluster-samples-operator/pull/249, where we add a ref to this proposal and delete any repeat information, we should probably remove all these CSI factoids and simply add a ref to https://github.com/openshift/cluster-samples-operator/pull/249, no?

File: enhancements/builds/entitlements.md
Comment (gabemontero) [inline enhancements/builds/entitlements.md]:
```suggestion
The build controller in turn would supply the annotation(s) the admission apparatus respects
```

File: enhancements/builds/entitlements.md
Comment (gabemontero) [inline enhancements/builds/entitlements.md]:
```suggestion
2. CRI-O already has config options (see the [config file doc](https://github.com/cri-o/cri-o/blob/master/docs/crio.conf.5.md))  to specify host directories that should always be mounted into containers (i.e. `/etc/containers/mounts.conf` to override the default `/usr/share/containers/mounts.conf`), akin to what the Red Hat Docker Daemon used to provide with OpenShift V3.  Guidance in the early 4.x time frame has been around the cluster admin using the MCO post install to effectively update the filesystem of each node with the necessary subscription related files in a well known location that CRI-O would look for.  There are some concerns though about using MCO for this purpose.  One, the MCO api is not natural for someone who wants to do something higher level like "enable entitlements for builds".  Second, using the MCO for any node update requires a restart of the node, which is onerous for this scenario.  Also note, the RHCOS team wants to disable their current default setting of auto-mounting secrets from the host (i.e. creating an empty `/etc/containers/mounts.conf`), since RHCOS does not fully install subscription manager (and never will), and they have problems when only partial subscription manager metadata is available.  [RHCOS has the subscription-manager-certs installed but are missing the subscription-manager package which carries the rhsm.conf.](https://bugzilla.redhat.com/show_bug.cgi?id=1783393)
```

**Review by gabemontero — CHANGES_REQUESTED** [review]
Just a few grammar/wording suggestions via suggested commits, as well as a call to remove CSI driver factoids and just reference the new PR (akin to removing broader scenario elements from the CSI driver PR and just reference this proposal).

Thanks for this rework and accelerating this along @bparees

File: enhancements/builds/entitlements.md
Comment (bparees) [reply enhancements/builds/entitlements.md:212]: yeah probably, can i leave that as a cleanup for you after you merge this?  you have a better understanding of CSI than me at this point and are probably better positioned to decide what minimal info needs to be provided here, vs what to defer to the lower level CSI EP.

**Comment by bparees** [comment]
@gabemontero updates pushed other than dropping the CSI lifecycle bits

**Comment by gabemontero** [comment]
> @gabemontero updates pushed other than dropping the CSI lifecycle bits

yep was just about to ask about that ;-) ... thanks @bparees

File: enhancements/builds/entitlements.md
Comment (gabemontero) [reply enhancements/builds/entitlements.md:212]: yep that works for me

**Pull request merged**
